[Frontend] TAPP-116: create shared file-related functions for components

michellemtchai opened the pull request:
Please refer to issue #116 for details.

Tested in the following browsers:
- [x] Firefox
- [x] Chrome
- [x] Edge
- [ ] IE  `// needs polyfill for fetch, promise, arrow function`
- [ ] Safari

As a starter, a file called `fileManager.js` is made for action involving files, such as:
 - `readFile`: a function that takes an input file and reads it as either text or JSON
 - `downloadFile`: a function that fetches a blob given at a certain route and triggers a download file response

Note: both of these functions output a JSON, using the second function argument, with the following format:
```
{
  success: boolean,
  content: string OR JSON, // content of file OR error message
}
```
Furthermore, both functions takes in a function as the second parameter. The function is assumed to take in only one argument. This single argument should be the aforementioned JSON.

Here is an example on the use of `readFile()` in HTML:
```
<input type='file' onchange='readFile(this, console.log)' />
```
Please note that the use of `onchange` is very important for getting the content of the file when you click on the upload button.

Here is an example on the use of `downloadFile()` in HTML:
```
<button onclick='downloadFile("https://www.wikipedia.org/portal/wikipedia.org/assets/img/sprite-556af1a5.svg", console.log)'>Download</button>
```
For a quick test of these functions, you can make a HTML like this:
```
<!DOCTYPE html>
<html>
<head>
	<script type="text/javascript" src="./fileManager.js"></script>
</head>
<body>
	<p>Upload Test</p>
	<input type='file' onchange='readFile(this, console.log)' />
	<button onclick='downloadFile("https://www.wikipedia.org/portal/wikipedia.org/assets/img/sprite-556af1a5.svg", console.log)'>Download</button>
</body>
</html>
```
Note: You will need to put this file into `/frontend/src/lib/` for it to test it on your browser. You will also need to comment out the `export` before the functions in `fileManager.js` because it's an independent JS file.

michellemtchai commented:
Please check if this code works on Safari if you have a Mac.

dingtaoliu commented on frontend/src/lib/fileManager.js:
```diff
@@ -0,0 +1,83 @@
+        reader.readAsText(files[0]);
+        reader.onload = event => importFunc(event.target.result);
+        component.value = '';
+    } else {
```
We can pull component.value out of the if block and get rid of the else to be more concise

michellemtchai replied:
I've fixed the redundancy. Thanks.

dingtaoliu reviewed (changes requested):
Just one small change, everything else looks good!  I like how generalized these functions are, good work!

michellemtchai committed:
feat(frontend): create filemanager.js in /frontend/src/lib
create a file for shared function across components

readFile: a function that takes an input file and reads it as either text or JSON

downloadFile: a function that fetches a blob given at a certain route and triggers a download file response
